packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -483,30 +478,142 @@ export class PerformanceEvent {
+    private getOwnValidTelemetryProps(): ITaggableTelemetryProperties {
+        const props: ITaggableTelemetryProperties = {};
+        for (const key of Object.keys(this)) {
+            const val = this[key];
+            switch (typeof val) {
+                case "string":
+                case "number":
+                case "boolean":
+                case "undefined":
+                    props[key] = val;
+                    break;
+                default: {
+                    if (IsTaggedTelemetryPropertyValue(val)) {
+                        props[key] = val;
+                    }
+                    break;
+                }
+            }
+        }
+        return props;
+    }
+
+    /**
+     * Get all properties fit to be logged to telemetry for this error
+     */
     public getTelemetryProperties(): ITelemetryProperties {
-        const props: ITelemetryProperties = {};
-        // Could not use {...this} because it does not return properties of base class.
-        for (const key of Object.getOwnPropertyNames(this)) {
-            props[key] = this[key];
+        // Copy non-enumerable props inherited from Error
+        const props: ITelemetryProperties = {
+            stack: this.stack,
+            message: this.message,
+            name: this.name,
+        };
+
+        // Walk the object's telemetry props, respecting the tags and converting them to ITelemetryProperties
+        // Note: Once ITelemetryBaseLogger is updated to use ITaggableTelemetryProperties,
+        // this loop will go away and we can just return the taggable properties directly for the host to deal with
+        const taggableProps = this.getOwnValidTelemetryProps();
+        for (const key of Object.keys(taggableProps)) {
+            const taggableProp = taggableProps[key];
+            const { value, tag } = (typeof taggableProp === "object")
+                ? taggableProp
+                : { value: taggableProp, tag: TelemetryDataTag.None };
+            switch (tag) {
+                case TelemetryDataTag.None:
+                case TelemetryDataTag.PackageData:
+                    // For Microsoft applications, PackageData is safe for now (we don't load 3P code in 1P apps)
+                    // But this determination really belongs in the host layer
+                    props[key] = value;
+                    break;
+                case TelemetryDataTag.UserData:
+                    // Strip out anything tagged explicitly as PII. Alternate strategy would be to hash these props
+                    props[key] = undefined;
+                    break;
+                default:
+                    // This will help us keep this switch statement up to date
+                    (function(_: never) {})(tag);
+
+                    // If we encounter a tag we don't recognize (e.g. due to interaction between different versions)
+                    // then we must assume we should scrub.
+                    props[key] = undefined;
```

Comment:
this is the only place we scrub, which is weird, why scrub at all

Reply:
it also strange that the error scrubs itself. seems like something the logger should do, and the root logger at that

---
In reply to: [593325577](https://github.com/microsoft/FluidFramework/pull/5308#discussion_r593325577) [](ancestors = 593325577)

Reply:
as i'd still want these logged at the console, even the user data

---
In reply to: [593326547](https://github.com/microsoft/FluidFramework/pull/5308#discussion_r593326547) [](ancestors = 593326547,593325577)

Reply:
> as i'd still want these logged at the console, even the user data
I'll open an issue, totally agree, but that will be an incremental change.  Good news is that nothing is actually being scrubbed since `UserData` isn't used yet.